scripts/write_srl_predictions_to_conll_format.py

```diff
@@ -1,40 +1,62 @@
 import os
 import sys
 
-import torch
-
 sys.path.insert(0, os.path.dirname(os.path.abspath(os.path.join(__file__, os.pardir))))
 import argparse
-from allennlp.common import Params
+
 from allennlp.common.tqdm import Tqdm
+from allennlp.common import Params
+from allennlp.models.archival import load_archive
 from allennlp.data.iterators import BasicIterator
 from allennlp.data import DatasetReader
 from allennlp.models import Model
 from allennlp.models.semantic_role_labeler import write_to_conll_eval_file
+from allennlp.modules.elmo import Elmo
 
-
-def main(serialization_directory, device):
+def main(serialization_directory: int,
+         device: int,
+         data: str,
+         prefix: str,
+         domain: str = None):
     """
     serialization_directory : str, required.
-        The directory containing the serialized weights.
+        the directory containing the serialized weights.
     device: int, default = -1
-        The device to run the evaluation on.
+        the device to run the evaluation on.
+    data: str, default = None
+        The data to evaluate on. By default, we use the validation data from
+        the original experiment.
+    prefix: str, default=""
+        The prefix to prepend to the generated gold and prediction files, to distinguish
+        different models/data.
+    domain: str, optional (default = None)
+        If passed, filters the ontonotes evaluation/test dataset to only contain the
+        specified domain.
     """
-    torch.set_grad_enabled(False)
-
     config = Params.from_file(os.path.join(serialization_directory, "config.json"))
+
+    if domain is not None:
+       config["dataset_reader"]["domain_identifier"] = domain
+       prefix = f"{domain}_{prefix}"
+
+    else:
+       config["dataset_reader"].pop("domain_identifier", None)
+
     dataset_reader = DatasetReader.from_params(config['dataset_reader'])
-    evaluation_data_path = config['validation_data_path']
 
-    model = Model.load(config, serialization_dir=serialization_directory, cuda_device=device)
+    evaluation_data_path = data if data else config['validation_data_path']
 
-    prediction_file_path = os.path.join(serialization_directory, "predictions.txt")
-    gold_file_path = os.path.join(serialization_directory, "gold.txt")
+    archive = load_archive(os.path.join(serialization_directory, "model.tar.gz"), cuda_device=device)
+    model = archive.model
+    model.eval()
+
+    prediction_file_path = os.path.join(serialization_directory, prefix + "_predictions.txt")
```

Comment:
Suggestion: if prefix is empty, or already ends with an underscore, this behaves a little oddly. You could make that nicer.

Reply:
The second argument is a string concatenation, rather than multiple arguments to os.join, so I don't think it does? It would just be "/_predictions.txt"

Reply:
Yes, and `/_predictions.txt` is what I meant by "behaves a little oddly" - the beginning underscore is unnecessary.

Reply:
I was doing the review from a tablet yesterday, so I didn't want to type out the whole thing, but I was thinking something like:

```python
if prefix and prefix[-1] != '_':
  prefix += '_'
```

Then you don't need to add an underscore at the beginning of the filename.

Again, this really isn't a big deal; it's up to you.